onecodex/models/collection.py

```diff
@@ -501,40 +472,83 @@ def _collate_results(self, metric=None, include_host=None):
             copy=False,
         )
 
-        self._cached["results"] = df
-        self._cached["taxonomy"] = tax_info
+        return df, tax_info
 
     @property
-    def _metric(self):
-        if "metric" not in self._cached:
-            self._collate_results()
+    def automatic_metric(self) -> Metric:
+        """
+        Return the best classification metric given the set of classifications.
+
+        Automatically determines the best metric to use. If more than half of the samples have
+        abundance estimates, use AbundanceWChildren. Otherwise, use ReadcountWChildren.
+        """
+
+        if getattr(self, "_automatic_metric", None):
+            return self._automatic_metric
+
+        metric_counts = {"abundance": 0, "readcount": 0, "readcount_w_children": 0}
+
+        for classification in self._classifications:
+            for row in classification.results()["table"]:
+                if row["rank"] != "species":
+                    continue
+                for key in metric_counts:
+                    if key in row and row[key] is not None:
+                        metric_counts[key] += 1
+
+        # if abundances are available in more than half of the samples, use that
+        if metric_counts["abundance"] >= (metric_counts["readcount_w_children"] // 2):
+            self._automatic_metric = Metric.AbundanceWChildren
+        else:
+            self._automatic_metric = Metric.ReadcountWChildren
 
-        return self._cached["metric"]
+        return self._automatic_metric
 
     @property
-    def _is_metagenomic(self):
-        if "is_metagenomic" not in self._cached:
-            # It looks like ._collate_results() does not actually populate 'is_metagenomic';
-            # should call _classification_fetch() instead?
-            self._collate_results()
+    def classification_ids_without_abundances(self) -> list[str]:
```

Comment:
Maybe this property should be "private"? `_classification_ids_without_abundances`